src/base.jl

```diff
@@ -232,6 +250,26 @@ for T in (UInt2048, UInt1024, UInt512, UInt256, UInt128, UInt64, UInt32)
     end
 end
 
+"""
+Return a ShortString type that can hold maxlen codeunits
+The keyword parameter `types` can be used to pass a list of types
+which can be used to store the string
+If no type is large enough, then an `ArgumentError` is thrown
+"""
+function get_type(maxlen; types=def_types)
+    for T in types
+        maxlen <= sizeof(T) - size_bytes(T) && return ShortString{T}
+    end
+    throw(ArgumentError("$maxlen is too large to fit into any of the provided types: $types"))
+end
+
+ShortString(str::Union{String,SubString{String}}, maxlen = 0; types=def_types) =
+    get_type(maxlen <= 0 ? sizeof(str) : maxlen, types=types)(str)
```

Comment:
```suggestion
ShortString(str::Union{String,SubString{String}}, maxlen = sizeof(str); types=def_types) =
    get_type(maxlen, types=types)(str)
```


If the user provides a maxlen of 0 or less it should error.